ang/test/civicase/CaseListDirectives.spec.js

```diff
@@ -151,18 +157,195 @@ describe('CaseListDirective', function () {
     });
 
     /**
-     * Mock jQuery scrollTop with given top value.
+     * Common setup for tests
      *
+     * @params {boolean} loading - wheather loading is complete
      * @params {Object} scope
      * @params {Number} top - px from top the screen should scroll to.
      */
-    function mockjQueryScrollTop (scope, top) {
+    function setupCommonSteps (loading, scope, top) {
+      scope.isLoading = loading;
       // Creating a custom function to mock offset() jQuery function
       CRM.$.fn.scrollTop = function () {
         return top;
       };
 
       scope.$digest();
+
+      if (!loading) {
+        $timeout.flush(); // Flushing any timeouts used.
+      }
+    }
+  });
+
+  describe('civicaseCaseListSortHeader directive', function () {
+    var element, $compile, $rootScope, scope, addClassOriginalFunction, toggleClassOriginalFunction, findOriginalFunction, removeOriginalFunction;
+
+    beforeEach(module('civicase'));
+
+    beforeEach(inject(function (_$compile_, _$rootScope_) {
+      $compile = _$compile_;
+      $rootScope = _$rootScope_;
+      scope = $rootScope.$new();
+    }));
+
+    beforeEach(function () {
+      addClassOriginalFunction = CRM.$.fn.addClass;
+      findOriginalFunction = CRM.$.fn.find;
+      removeOriginalFunction = CRM.$.fn.remove;
+      toggleClassOriginalFunction = CRM.$.fn.toggleClass;
+
+      spyOn(CRM.$.fn, 'addClass').and.callThrough();
+      spyOn(CRM.$.fn, 'find').and.callThrough();
+      spyOn(CRM.$.fn, 'remove').and.callThrough();
+      spyOn(CRM.$.fn, 'toggleClass').and.callThrough();
+    });
+
+    afterEach(function () {
+      CRM.$.fn.addClass = addClassOriginalFunction;
+      CRM.$.fn.find = findOriginalFunction;
+      CRM.$.fn.remove = removeOriginalFunction;
+      CRM.$.fn.toggleClass = toggleClassOriginalFunction;
+    });
+
+    describe('basic tests', function () {
+      var header;
+
+      describe('when sortable is false', function () {
+        beforeEach(function () {
+          scope.sort = { field: '', dir: '', sortable: false };
+          header = {
+            display_type: 'activity_card',
+            label: 'Next Activity',
+            name: 'next_activity',
+            sort: 'next_activity'
+          };
+          compileDirective(scope, header);
+        });
+
+        it('does not make the header sortable', function () {
+          expect(CRM.$.fn.addClass).not.toHaveBeenCalledWith('civicase__case-list-sortable-header');
```

Comment:
`expect(element.hasClass('civicase__case-list-sortable-header')).toBe(true);` but this is fine too. My issue is that you need to know a lot about jQuery internals to do this kind of tests.